src/main/java/seedu/address/logic/parser/cookbook/CookbookCommandParser.java

```diff
@@ -9,12 +19,38 @@
  */
 public class CookbookCommandParser implements Parser<CookbookCommand> {
 
+    private static final Pattern COOKBOOK_COMMAND_ARGUMENT_FORMAT = Pattern
+            .compile("(?<commandWord>\\S+) (?<index>\\S+)");
+    private static final Pattern COOKBOOK_ADD_COMMAND_ARGUMENT_FORMAT = Pattern
+            .compile("(?<commandWord>\\S+) (?<arguments>.*)");
+
     /**
      * Parses the given {@code String} of arguments in the context of a CookbookCommand
      * and returns a CookbookCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public CookbookCommand parse(String args) throws ParseException {
-        throw new ParseException("Not implemented yet");
+        final Matcher matcher1 = COOKBOOK_COMMAND_ARGUMENT_FORMAT.matcher(args.trim());
+        final Matcher matcher2 = COOKBOOK_ADD_COMMAND_ARGUMENT_FORMAT.matcher(args.trim());
+
+        if (!matcher1.matches() && !matcher2.matches()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+        }
+
+        final String commandWord = matcher1.group("commandWord");
+        final String index = matcher1.group("index");
+
+        final String arguments = matcher2.group("arguments");
+
+        switch(commandWord) {
+        case CookbookAddCommand.COMMAND_WORD:
+            return new CookbookAddCommandParser().parse(arguments);
+        case CookbookViewCommand.COMMAND_WORD:
+            return new CookbookViewCommandParser().parse(index);
+        case CookbookRemoveCommand.COMMAND_WORD:
+            return new CookbookRemoveCommandParser().parse(commandWord + " " + index);
+        default:
+            throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
+        }
```

Comment:
so then each case would be the new `XxxParser.parse(arguments)` which can handle the indices/prefixes themselves respectively

Reply:
no need to change the existing parsers, cus they already handle the indices/prefixes fine.